lib/network/encoding.go

```diff
@@ -66,6 +68,24 @@ func TypeFromData(msg ProtocolMessage) uuid.UUID {
 	return msgType
 }
 
+// TypeToUUID Converts MsgType to uuid
+func TypeToUUID(msg ProtocolMessage) uuid.UUID {
+	val := reflect.ValueOf(msg)
+	if val.Kind() == reflect.Ptr {
+		val = val.Elem()
+	}
+	url := "https://dedis.epfl.ch/protocolType/" + val.Type().String()
```

Comment:
Nitpicking: Maybe a constant with the base url would be better? This would make sure that 1) The bas url can easily be changed and 2) it's the same everywhere (no typos).